pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -371,45 +370,44 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 		allNodeNames = append(allNodeNames, name)
 	}
 
-	var lock sync.Mutex
-	var firstError error
-
-	topologyMaps := newTopologyPairsMaps()
-
-	appendTopologyPairsMaps := func(toAppend *topologyPairsMaps) {
-		lock.Lock()
-		defer lock.Unlock()
-		topologyMaps.appendMaps(toAppend)
-	}
-	catchError := func(err error) {
-		lock.Lock()
-		defer lock.Unlock()
-		if firstError == nil {
-			firstError = err
-		}
-	}
+	allNodesCount := len(allNodeNames)
+	errCh := make(chan error, 1)
+	topologyPairsCh := make(chan *topologyPairsMaps, allNodesCount)
```

Comment:
This looks pretty a lot of memory. I'm thinking of a way to limit the buffer size (like 16), and use a producer-consumer model to run a background consumer goroutine to consume it simultaneously. And for sure this can be in a separate PR and needs benchmarking.

Reply:
> This looks pretty a lot of memory. I'm thinking of a way to limit the buffer size (like 16), and use a producer-consumer model to run a background consumer goroutine to consume it simultaneously.

Sounds good, but it may cause the scheduler schedules the goroutine back and forth.

> And for sure this can be in a separate PR and needs benchmarking.

Sure, will do that.

Reply:
One way to move forward is to have this PR focused only on cleaning up the error handling since memory concern is not an issue for that channel, and make the other changes to a separate PR that also provide benchmarking numbers.

Reply:
Opened an issue https://github.com/kubernetes/kubernetes/issues/79891, and will run the benchmark it later.